pkg/didcomm/common/service/destination.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package service
+
+import (
+	diddoc "github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdri"
+)
+
+// Destination provides the recipientKeys, routingKeys, and serviceEndpoint for an outbound message.
+// Can be populated from an Invitation or DIDDoc.
+type Destination struct {
+	RecipientKeys        []string
+	ServiceEndpoint      string
+	RoutingKeys          []string
+	TransportReturnRoute string
+}
+
+// GetDestination constructs a Destination struct based on the given DID and parameters
+// It resolves the DID using the given VDR, and collects relevant data from the resolved DIDDoc.
+func GetDestination(did, serviceType, keyType string, vdr vdri.Registry) (*Destination, error) {
+	didDoc, err := vdr.Resolve(did)
+	if err != nil {
+		return nil, err
+	}
+
+	return CreateDestination(didDoc, serviceType, keyType)
+}
+
+// CreateDestination makes a Destination object from a DID Doc, following the given parameters.
+func CreateDestination(didDoc *diddoc.Doc, serviceType, keyType string) (*Destination, error) {
```

Comment:
`serviceType` should be removed. It's always set to `did-communication`.

Reply:
`keyType` should also be removed as an arg.

The keyType was ensured to be `Ed25519VerificationKey2018` in the `didexchange` package because of the crypto primitives we were willing to support in the framework. Callers of this function should not be allowed to specify their own keyTypes.

Reply:
> It's always set to did-communication.
Always, for any communication using a Destination created from a DID doc?
Sounds right but just want to be sure.

keyType will need to be refactored away when we have a mechanism for selecting key type. Should I just enforce it to be that constant for now, in here?

Reply:
> Always, for any communication using a Destination created from a DID doc?

Yes. See [Service Conventions](https://github.com/hyperledger/aries-rfcs/blob/master/features/0067-didcomm-diddoc-conventions/README.md#service-conventions).

> keyType will need to be refactored away when we have a mechanism for selecting key type. Should I just enforce it to be that constant for now, in here?

Moving the enforcement to this centralized location makes sense to me.

Reply:
done